src/lib/methods/delete-item.ts

```diff
@@ -81,7 +95,16 @@ export class DeleteItem extends Method implements Executable {
 			return Promise.reject(new Error('Call .connect() before executing queries.'));
 		}
 
-		return db.delete(this.buildRawQuery()).promise()
+		/**
+		 * Use the retry options configured for operation if defined
+		 * Fallback to the retry options declared when db was initialized if defined
+		 * If none of the above is defined, no retry mechanic is used
+		 */
+		const queryResult = this.retryOptions || this.dynamodb.retryOptions
+			? retry(() => db.delete(this.buildRawQuery()).promise().catch(retryErrorHandler), this.retryOptions)
```

Comment:
Can we use `async/await` here instead?

I'm wondering if we can make all of this a util function?

```ts
const queryResult = runQuery(() => {
	return db.delete(this.buildRawQuery()).promise();
}, this.retryOptions || this.dynamodb.retryOptions);
```

The implementation picks the execution based on the input of the `retryOptions`.

Reply:
Made a method in the `Method` class that can execute with or without retry.